api/v1alpha1/circuitbreaker_types.go

```diff
@@ -16,28 +16,35 @@ type CircuitBreakers struct {
 }
 
 type Thresholds struct {
-	// The maximum number of connections that Envoy will make to the referenced backend (per xRoute).
-	// Default: 1024
+	// The maximum number of connections that Envoy will establish to the referenced backend (per xRoute).
 	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=4294967295
+	// +kubebuilder:default=1024
 	// +optional
 	MaxConnections *uint32 `json:"maxConnections,omitempty"`
 
-	// The maximum number of pending requests that Envoy will allow to the referenced backend (per xRoute).
-	// Default: 1024
+	// The maximum number of pending requests that Envoy will queue to the referenced backend (per xRoute).
 	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=4294967295
```

Comment:
why is `4294967295` ?

Reply:
this is the maximum value of `uint32`, but i think this `Maximum` validation can be optional

Reply:
oh, my bad, think about `2147483647`.

Reply:
can `-1` pass if the type is uint32?

Reply:
no, it cannot

Reply:
so these rules seem meaningless?

Reply:
`Maximum` and `Minimum` sure seems meaningless, but can keep `default`, WDYT?

Reply:
Yes, that's what it means.

Reply:
Hi @zirain, @shawnh2.  Do note that the OpenAPI spec (used by K8s CRDs) doesn't really support unsigned ints: https://swagger.io/specification/. The controller-gen tools actually produce a schema that refers to these fields as `int32` in the generated CRD. The actual K8s API server behavior, from my limited check, is to treat these fields as `int64`. I think that the actual go type (`*uint32`) mostly impacts the unmarshalling done by client go. So, guaranteeing that the value stored is actually safe to cast to uint32 could be useful...

Reply:
Another approach would be to use `int64` explicitly in the go types layer and have uint32 as a representation in the IR layer and downwards. The value range validation can occur either using the schema or during the IR translation. WDYT?

Reply:
like Gateway API project, let's use `*int32` with valiation min and max?

Reply:
Wouldn't it make more sense to use `*int64`? `MaxUInt32` > `MaxInt32`, so by using `*int32` users would not able to use the full value range provided by Envoy.